### PR TITLE
fix(NODE-3921): error on invalid TLS option combinations

### PR DESCRIPTION
**Pull request by biniona-mongodb**
### Description

Unskip tests to ensure that specifying mutually exclusive TLS options raises an error.

#### What is changing?

- Update TLS check to use `allOptions` object rather than `mongoOptions` object. This was done as `mongoOptions` object does not contain options with falsy values.
- Move TLS check before population of `mongoOptions` object.
- Remove export of function not used outside module.

##### Is there new documentation needed for these changes?

None

#### What is the motivation for this change?

https://jira.mongodb.org/browse/NODE-3921

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the correct format: `<type>(NODE-xxxx)<!>: <description>`
- [x] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: src/connection_string.ts [hunk 1]
```diff
@@ -166,14 +165,14 @@ export function resolveSRVRecord(options: MongoOptions, callback: Callback<HostA
...
  */
-export function checkTLSOptions(options: AnyOptions): void {
-  if (!options) return;
+function checkTLSOptions(allOptions: CaseInsensitiveMap): void {
+  if (!allOptions) return;
   const check = (a: string, b: string) => {
-    if (Reflect.has(options, a) && Reflect.has(options, b)) {
-      throw new MongoParseError(`The '${a}' option cannot be used with '${b}'`);
+    if (allOptions.has(a) && allOptions.has(b)) {
+      throw new MongoAPIError(`The '${a}' option cannot be used with the '${b}' option`);
```
Comment (biniona-mongodb) [inline src/connection_string.ts:175]: See [this thread](https://github.com/mongodb/node-mongodb-native/pull/3397#discussion_r963760291) for context on why this error type was changed

File: src/connection_string.ts [hunk 2]
```diff
@@ -369,6 +368,8 @@ export function parseOptions(
     }
   }
 
+  checkTLSOptions(allOptions);
```
Comment (biniona-mongodb) [inline src/connection_string.ts:371]: Update check from `mongoOptions` to `allOptions` as `mongoOptions` object does not have key value pairs [when value would be falsy](https://github.com/mongodb/node-mongodb-native/blob/8b0ba59b4142237adaf19fa5bbe8c6195d2e34a0/src/connection_string.ts#L389).

File: src/connection_string.ts [hunk 3]
```diff
@@ -166,14 +165,14 @@ export function resolveSRVRecord(options: MongoOptions, callback: Callback<HostA
 /**
  * Checks if TLS options are valid
  *
- * @param options - The options used for options parsing
- * @throws MongoParseError if TLS options are invalid
+ * @param allOptions - all options provided by user
+ * @throws MongoAPIError if TLS options are invalid
  */
-export function checkTLSOptions(options: AnyOptions): void {
```
Comment (biniona-mongodb) [inline src/connection_string.ts:172]: Remove `export` because AFAICT `checkTLSOptions` is not used elsewhere in driver and is not included in public API.

**Review by nbbeeken — DISMISSED** [review]
LGTM, Thanks for the help here!

**Comment by baileympearson** [comment]
hey @biniona-mongodb - thanks for the fix.  Once the merge conflicts on this PR have been resolved, this should be good go to.

if you don't have time, we can always resolve the conflicts as well.  just let us know

**Comment by biniona-mongodb** [comment]
Hey @baileympearson, I just resolved the conflicts. Let me know if you'd like to see any additional changes 🙂.

**Review by baileympearson — APPROVED** [review]
@nbbeeken I logged the flaky tests and reran the failures, CI is green now

**Review by nbbeeken — APPROVED** [review]

**Pull request merged**